controller/build/build.go

```diff
@@ -379,3 +382,19 @@ func controllerUlimitOpt2DockerUlimit(u *controllerapi.UlimitOpt) *dockeropts.Ul
 	}
 	return dockeropts.NewUlimitOpt(&values)
 }
+
+type ResultContextError struct {
+	ResultContext *build.ResultContext
+	error
+}
```

Comment:
Not sure I understand the purpose of this - doesn't `build.ResultContext` already contain an error type?

Reply:
Added explanation of this struct as a comment in the code.

```
// ResultContextError is an error type used for passing ResultContext from this package
// to the caller of RunBuild. This is only used when RunBuild fails with an error.
// When it succeeds without error, ResultContext is returned via non-error returned value.
//
// Caller can extract ResultContext from the error returned by RunBuild as the following:
//
//	resp, res, buildErr := cbuild.RunBuild(ctx, req.Options, inR, statusChan)
//	var re *cbuild.ResultContextError
//	if errors.As(buildErr, &re) && re.ResultContext != nil {
//		res = re.ResultContext
//	}
```

Reply:
Thanks for the comment. Fixed the code not to use ResultContextError.